ssh/tailssh/incubator.go

```diff
@@ -204,6 +204,11 @@ func parseIncubatorArgs(args []string) (a incubatorArgs) {
 // OS, sets its UID and groups to the specified `--uid`, `--gid` and
 // `--groups` and then launches the requested `--cmd`.
 func beIncubator(args []string) error {
+	// To defend against issues like https://golang.org/issue/1435,
+	// defensively lock our current goroutine's thread to the current
+	// system thread before we start making any UID/GID/group changes.
+	runtime.LockOSThread()
```

Comment:
Do we need to do this in an `init()`? And if this method is hit by tests, do we need to do a `defer runtime.UnlockOSThread()` to be good thread denizens?

Reply:
I don't think we can do this in an `init`, due to the way that the childproc stuff works in Tailscale. I did add a defer unlock though; even though this isn't called in tests, no reason not to 👍